[feat](#2): kakao Login

wisdomin121 opened the pull request:
### 📃 전달 사항

백엔드 서버에서 토큰 받아서 쿠키에 저장하는 것 까지 완료 !

hooks/queries/checkId 에서 백엔드 서버에 사용자 정보 보낼 수 있고,
components/layout/KakaoLayout 에 있는 코드와 같이 결과 값 받아와서 
hooks/useSaveToken 에서 토큰 저장할 수 있게 해놨습니당

애플 로그인에서도 재사용 할 수 있게 만들긴 했는데, 보내는 게 다르다고 들어서 코드 살짝 바꿔서 사용해야 할 것 같습니다..!

### 📸 스크린샷
![image](https://user-images.githubusercontent.com/46989954/229385643-6c8e35d4-7d1f-4b7b-b20c-c57746ef73d6.png)

### ✔ 진행사항

 - [x] 카카오 서버에 인가코드 넘겨주고 토큰 받기
 - [x] 카카오 서버에 토큰 넘겨주고 사용자 정보 받기
 - [x] 백엔드 서버에 사용자 정보 넘겨주고 토큰 받기
 - [x] refresh token, access token 저장하기

### 🔴 ETC

기타 사항을 적어주세요.
실제 개발 기간 : 3주

<hr>

closes: #2

wisdomin121 committed:
[chore]: prettier printWidth 변경

변경 사항 : printWidth를 80 -> 120 으로
변경 이유 : 80이 기본 값이긴 하지만, prettier 공식 문서를 확인했을 때, 가독성을 위해 100이나 120으로 설정할 것을 추천해서 바꿨습니다.

wisdomin121 committed:
[feat]: 카카오 서버에 인가코드 넘겨주고 토큰 받기

테스트 버튼 만들어서 인가코드 받아오고, 넘겨주는 이벤트 추가

wisdomin121 committed:
[feat]: 카카오 서버에 토큰 넘겨주고 사용자 정보 받기

- axios 인스턴스 생성 (/configs/axios.ts KAKAO_USER_INFO_API)
- api 생성 (apis/userApis.ts -> kakaoUserInfo)
- query 생성 (hooks/queries/kakao.ts -> useKakaoUserInfoQuery)
- 함수 생성 (configs/kakao.ts -> getKakaoUserInfo & saveToken)

Raoun4136 commented on src/apis/userApi.ts:
```diff
@@ -11,15 +11,34 @@ export interface ILoginData {
     const data = await API.get('/')
     return data
+  },
+  kakaoToken: async (data: IKakaoTokenData) => {
```
API 이름이 KaKaoToken 이면 apis.KaKaoToken일텐데, 동사로 쓰는게 낫지 않을까? getKaKaoToken 이라던지

wisdomin121 replied:
아 동사가 맞는 것 같아 굳굳 감사합니다ㅏ

Raoun4136 commented on src/configs/kakao.ts:
```diff
@@ -0,0 +1,37 @@
+  setCookie('refreshToken', refreshToken)
+}
+
+const getKakaoUserInfo = () => {
```
getKaKao000 들이 config에 들어갈 내용인가 모르겠어
아예 user hook쪽에 박아놓는것도 좋을 것 같은데
cookie설정도 hook과 apis.~~들에 같이 넣어서 설정하면 더 좋을 것 같아

Raoun4136 commented on src/hooks/queries/kakao.ts:
```diff
@@ -0,0 +1,43 @@
+  return useQuery(
+    'kakaoToken',
+    async () => {
+      return await apis.kakaoToken({
```
위에도 말했듯이 getKaKaoToken이 더 가독성 있어보여서 ㅎㅎ 근데 그러면 config에 있는 거랑 중복이네...

그리구
``` typescript
return await apis.kakaoToken
```
이 부분 오류나더라 await 쓰지 말라고 하던데

wisdomin121 replied:
헉 확인해볼게 !

Raoun4136 commented on src/components/layout/KakaoLayout.tsx:
```diff
@@ -0,0 +1,13 @@
+const KakaoLayout = () => {
+  const AUTHORIZATION_CODE: string = new URL(document.location.toString()).searchParams.get('code') as string
+
+  getKakaoToken(AUTHORIZATION_CODE)
```
카카오 레이아웃은 나중에 모달창 생기면 어짜피 사라지는 거지?.?

wisdomin121 replied:
인가 코드를 Redirect URI로 보내주는 데, 코드를 빼오려면 Redirect URI에 맞는 빈 페이지가 필요해서 만든 거라서,, 아마 계속 있을 것 같아 !!

혹시 페이지 안 만들고 빼올 수 있는 방법이 있나 ??

Raoun4136 commented on src/pages/landing/LandingPage.tsx:
```diff
@@ -14,6 +16,14 @@ const LoginPage = () => {
+
+        <button
+          onClick={() => {
+            getKakaoAuthorizationCode()
```
지금은 코드받아서 넘기는 것 밖에 못하지만 나중에 KaKaoLoginMutation도 추가가 되면 코드 하나 실행으로 훅-> api 거쳐서 로그인 다 진행될 수 있게끔... 하면 좋겠지만 나도 처음이라 잘 모르겠다 ㅋㅋㅋ

Raoun4136 reviewed (commented):
리뷰 완료했습니다! 아직 머지할거 아닌 것 같아서 일단 comment로만 남겨둘게!

wisdomin121 committed:
[style]: api이름 동사로 수정

wisdomin121 committed:
[fix]: 카카오에서 발급된 accessToken이 넘어가지 않았던 오류 해결

쿼리를 두 개 사용했을 때,
첫 번째 쿼리와 두 번째 쿼리가 같이 실행되어
accessToken이 두 번째 쿼리로 넘어가지 않음.

현재는 useQueries를 사용하여
두 개의 쿼리를 동기적으로 사용할 수 있게 변경함

wisdomin121 committed:
[fix]: 카카오 redirect URI 경로 변경

백엔드와 경로가 같지 않으면,
오류가 생길 수 있어서 사전에 변경함

BEFORE
  Kakao = '/kakao'

AFTER
  Kakao = '/oauth/kakao'

wisdomin121 committed:
[feat]: 백엔드 서버에 사용자 정보 넘겨주고 토큰 받기

- hooks/checkId.ts : 애플 로그인과 동일한 코드로 진행될 예정이라
  카카오 쿼리에 포함시키지 않고 따로 빼둠

wisdomin121 committed:
[feat]: 백엔드 서버에서 받아온 토큰 저장

- useSaveToken : 토큰을 쿠키로 저장하는 코드
  ( 애플 로그인에서도 사용할 것 같아 hook으로 만들어둠)

wisdomin121 committed:
[refactor]: getKakaoAuthorizationCode 함수 삭제

configs/kakao에 있을 함수가 아니어서 삭제하고,
직접 입력함

BEFORE
  onClick={() => {
    getKakaoAuthorizationCode()
  }}

After
  onClick={() => {
    window.location.href = KAKAO_AUTH_URL
  }}

Raoun4136 committed:
[feat]: 메인 애니메이션 & 라우터 적용

Raoun4136 committed:
Merge branch 'feature/#2/kakao_login' of github.com:telling-me/tellingme-client into feature/#1/main_ui

wisdomin121 committed:
[style]: 스타일 가이드 수정본 반영

wisdomin121 committed:
[feat]; 색상 변경 함수 생성

primary200, error400 등 string을 넣어주면
해당하는 HEX 값을 반환해주는 함수를 생성함

wisdomin121 committed:
[feat]: 글자 크기 변경 함수 생성

h1, h2, h2_b 등 string을 넣어주면
해당하는 px값을 반환해주는 함수를 생성함

wisdomin121 committed:
[style]: type명 변경

ITextSize -> TextSizeType

인터페이스는 앞에 I를
타입은 뒤에 Type을 붙임

wisdomin121 committed:
[feat]: 아이콘 컴포넌트 생성

자세한 설명은 노션에 적어둠

wisdomin121 committed:
[feat]: 버튼 컴포넌트 생성

자세한 설명은 노션에 적어둠

wisdomin121 committed:
[feat]: 인풋 컴포넌트 생성

자세한 설명은 노션에 적어둠

wisdomin121 committed:
[style]: shadow 설정 수정

기존엔 'box-shadow: '이 포함되어서 쉐도우 적용이 안됐었음
현재는 지운 상태임

wisdomin121 committed:
[style]: caption 이름 변경

기존에 타입이 c1, c1_b로 되어 있어서
typo에 적어둔 caption 이름과 달랐음.
현재는 type을 c, c_b로 변경함

wisdomin121 committed:
[fix]: useChangeColor 조건 변경

beforeColor.includes('primary') 형식의 조건은 includes를 사용하지 못해서
Object.keys(Theme.colors.primary).includes(beforeColor) 형식의 조건으로 변경함

wisdomin121 committed:
[style]: TextSpan 간소화

전에 만들어둔 useChangeTextSize와 useChangeColor 훅으로
조건을 간소화시킴

wisdomin121 committed:
[feat]: 체크박스 컴포넌트 생성

자세한 설명은 노션에 적어둠

wisdomin121 committed:
[style]: 아이콘 컴포넌트 마진 속성 추가

wisdomin121 committed:
[style]: TextSpan 너비 속성 추가

wisdomin121 committed:
[feat]: 드롭다운 컴포넌트 생성

자세한 정보는 노션에 적어둠

wisdomin121 committed:
[hotfix]: Mixed Content 임시 해결책

+ [style]; 폰트 적용

http -> https 통신 에러 임시로 해결 코드 넣어둠

wisdomin121 committed:
[feat]: 회원가입 ui 생성

+[style]: 버튼, 체크박스, 드롭다운, 인풋 속성 추가

회원가입 ui 만들면서 컴포넌트 전체적으로 추가해야 할 속성들이 있어
추가하면서 진행함

wisdomin121 committed:
[feat]: 회원가입 통신 완료

+[style]: 버튼 withinputframe 타입 생성
  직업에서 '기타' 선택시 입력할 수 있는 버튼 생성함

wisdomin121 committed:
[style]: 프로그래스바 컴포넌트 생성

wisdomin121 committed:
[fix]: redirect uri 도메인 설정

로컬과 상용 모두 사용할 수 있도록 변경

wisdomin121 committed:
[hotfix]: MixedContent 해결 코드 추가

wisdomin121 committed:
[feat]: 추가 정보 닉네임 체크 기능 생성

Raoun4136 committed:
[feat]: apple socialId 받아오기 진행중

- 서버 https 보안 적용안됨 -> 받아온다는 가정하에 임시 url 생성

Raoun4136 committed:
Merge branch 'feature/#2/kakao_login' of github.com-raoun4136:telling-me/tellingme-client into feature/#4/apple_login

wisdomin121 committed:
[feat]: 추가 정보 알림 허용 여부 추가

+[style]: 아이콘 그라데이션 적용 가능하도록 수정
+[style]: 아이콘 그림자 적용 가능하도록 수정
+[feat]: 버튼 컴포넌트 배경 없는 벝튼 추가 (임시)

+[style]: 프로그래스바 컴포넌트 radius 추가
+[style]: 추가 정보 화면 화살표 버튼 짤리지 않도록 수정
+[style]: 추가 정보 화면 아이콘 shadow 추가
